### PR TITLE
Add plugin details to host catalog, host set, and host domain model pages

### PR DESCRIPTION
**Pull request by talanknight**

### CODE REVIEW

**Review by mgaffney — CHANGES_REQUESTED** [review]
Can remove the formatting only changes or split them into a different commit?

**Review by mgaffney — DISMISSED** [review]

**Review by sswati0 — APPROVED** [review]

File: website/content/docs/concepts/domain-model/host-catalogs.mdx [hunk 1]
```diff
@@ -20,6 +20,19 @@ A host catalog has the following configurable attributes:
...
 - `description` - (optional)
 
+### Plugin Host Catalog Information and Attributes
+
+- `plugin_id` - (required)
+  Must reference an installed plugin.
+
+- `attributes` - (optional)
+  A collection of fields which the plugin uses to interface with the backing
+  service. The set of fields which should be included in this attributes.
```
Comment (mgaffney) [inline website/content/docs/concepts/domain-model/host-catalogs.mdx:30]:
```
The set of fields which should be included in this attributes.
```
This sentence doesn't make sense to me.

File: website/content/docs/concepts/domain-model/host-sets.mdx [hunk 2]
```diff
@@ -25,6 +25,23 @@ A host set has the following configurable attributes:
 
 - `description` - (optional)
 
+- `preferred_endpoints` - (optional)
+  A list of selector strings in the format of either `cidr:<valid IPv4/6 CIDR>`
+  or `dns:<globbed name>` used to select the addresses of a [hosts][] when
```
Comment (mgaffney) [inline website/content/docs/concepts/domain-model/host-sets.mdx:30]:
```suggestion
  or `dns:<globbed name>` used to select the addresses of [hosts][] when
```

**Review by mgaffney — APPROVED** [review]
I left some minor suggestions.

**Review by mgaffney — APPROVED** [review]

**Pull request merged**